core/services/relay/types/relay.go

```diff
@@ -13,32 +13,55 @@ import (
 type Network string
 
 var (
-	EVM    Network = "evm"
-	Solana Network = "solana"
-	Terra  Network = "terra"
+	EVM             Network = "evm"
+	Solana          Network = "solana"
+	Terra           Network = "terra"
+	SupportedRelays         = map[Network]struct{}{
+		EVM:    {},
+		Solana: {},
+		Terra:  {},
+	}
 )
 
-// RelayerCtx represents a relayer
-type RelayerCtx interface {
+// PluginArgs are the args required to create any OCR2 plugin components.
+// Its possible that the plugin config might actually be different
+// per relay type, so we pass the config directly through.
+type PluginArgs struct {
+	ConfigWatcherArgs
+	TransmitterID string
+	PluginConfig  []byte
+}
+
+type ConfigWatcherArgs struct {
+	ExternalJobID uuid.UUID
+	JobID         int32
```

Comment:
Do we really need the `ExternalJobID` and `JobID` to watch config?

It feels that these are some base args that don't really fit here perfectly.

Reply:
my thinking was just it'd be desirable for logging, to link all the logs together